streams/src/test/java/org/apache/kafka/test/MockProcessorContext.java

```diff
@@ -299,4 +296,8 @@ public ProcessorNode currentNode() {
         return currentNode;
     }
 
+    public void close() {
```

Comment:
Ah right, we only create the metrics object in this mock processor context, and in practice metrics is closed following another path. Thanks for the explanation.